applications/matter_bridge/src/simulated_providers/simulated_temperature_sensor_data_provider.cpp

```diff
@@ -63,3 +67,10 @@ void SimulatedTemperatureSensorDataProvider::NotifyAttributeChange(intptr_t cont
 				    Clusters::TemperatureMeasurement::Attributes::MeasuredValue::Id,
 				    &provider->mTemperature, sizeof(provider->mTemperature));
 }
+
+CHIP_ERROR SimulatedTemperatureSensorDataProvider::SetValue(const void *newValue)
+{
+	VerifyOrReturnError(newValue, CHIP_ERROR_INCORRECT_STATE);
+	memcpy(&mTemperature, newValue, sizeof(mTemperature));
```

Comment:
Actually, I wonder why `mTemperature` is needed at all. The real state is held by the device, not provider object.

Reply:
This is a matter of how we perceive the provider, I guess the intuitive approach was to keep the state of data provider as such on the provider itself and forward data to the Matter side to reflect the behavior of real sensor or indicator. I theory these values could be read by another device. Pragmatically, I think it also makes interacting with `DeviceLayer::PlatformMgr().ScheduleWork()` API easier. @kkasperczyk-no  what's your opinion on that?

Reply:
Ah, I see. I didn't look carefully at how `ScheduleWork()` is used. It looks fine with me then.